Incorrect documentation example for client-server test

### Version

4.0.3

### Context

I recently encountered some occasional issues with testing client-server communication. I used the the test setup from [the documentation](https://vertx.io/docs/vertx-junit5/java/#_checkpoint_when_there_are_multiple_success_conditions) like:

```java
    Checkpoint serverStarted = testContext.checkpoint();
    Checkpoint requestsServed = testContext.checkpoint(10);
    Checkpoint responsesReceived = testContext.checkpoint(10);

    vertx.createHttpServer()
      .requestHandler(req -> {
        req.response().end("Ok");
        requestsServed.flag();
      })
      .listen(8888)
      .onComplete(testContext.succeeding(httpServer -> serverStarted.flag()));

    HttpClient client = vertx.createHttpClient();
    for (int i = 0; i < 10; i++) {
      client.request(HttpMethod.GET, 8888, "localhost", "/")
        .compose(req -> req.send().compose(HttpClientResponse::body))
        .onComplete(testContext.succeeding(buffer -> testContext.verify(() -> {
          assertThat(buffer.toString()).isEqualTo("Ok");
          responsesReceived.flag();
        })));
    }
```
When running those test there were some occasional failures with stack traces like:
```
io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:41057
Caused by: java.net.ConnectException: Connection refused
```
Unfortunately, most of the time those failures happened only on our CI system and I first assumed some firewall configuration problems. However, I think the code snippet above is wrong. The clients may send requests to the server while it is still starting up and cause the above mentioned stack trace. Moving all client calls in the `onComplete` of the server creation fixed the problem and I haven't encountered it since then (I assume that our CI system is just slower in processing events from the event loop). Therefore I propose to change the example above to the following:

```java
    Checkpoint serverStarted = testContext.checkpoint();
    Checkpoint requestsServed = testContext.checkpoint(10);
    Checkpoint responsesReceived = testContext.checkpoint(10);

    vertx.createHttpServer()
      .requestHandler(req -> {
        req.response().end("Ok");
        requestsServed.flag();
      })
      .listen(8888)
      .onComplete(testContext.succeeding(httpServer -> {
        serverStarted.flag();

        HttpClient client = vertx.createHttpClient();
        for (int i = 0; i < 10; i++) {
          client.request(HttpMethod.GET, 8888, "localhost", "/")
            .compose(req -> req.send().compose(HttpClientResponse::body))
            .onComplete(testContext.succeeding(buffer -> testContext.verify(() -> {
          assertThat(buffer.toString()).isEqualTo("Ok");
          responsesReceived.flag();
        })));
    }
       }));
```

### Do you have a reproducer?

[Vertx Junit5 Examples](https://github.com/vert-x3/vertx-junit5/blob/master/src/main/java/examples/Examples.java)
I think this might be related to https://github.com/vert-x3/vertx-junit5/issues/96 in a sense that the http server, started in `BeforeEach` hooks, are not yet up when the clients may already send requests. I assume this might occur more frequently on CI system that are slower in processing events in the event loop.

### Steps to reproduce

1. run those tests multiple times until a failure occurs (`while [ $? -eq 0 ]; do mvn clean test; done`).

### Extra

 I offer to submit a PR if this proposal is accepted

~ java -version
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment (build 11.0.10+9-Ubuntu-0ubuntu1.20.04)
OpenJDK 64-Bit Server VM (build 11.0.10+9-Ubuntu-0ubuntu1.20.04, mixed mode, sharing)

~ uname -a
Linux Colter 5.8.0-50-generic #56~20.04.1-Ubuntu SMP Mon Apr 12 21:46:35 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

--- a/src/main/java/examples/Examples.java
+++ b/src/main/java/examples/Examples.java
@@ -126,17 +126,19 @@ public class Examples {
         requestsServed.flag();
       })
       .listen(8888)
-      .onComplete(testContext.succeeding(httpServer -> serverStarted.flag()));
+      .onComplete(testContext.succeeding(httpServer -> {
+        serverStarted.flag();
 
-    HttpClient client = vertx.createHttpClient();
-    for (int i = 0; i < 10; i++) {
-      client.request(HttpMethod.GET, 8888, "localhost", "/")
-        .compose(req -> req.send().compose(HttpClientResponse::body))
-        .onComplete(testContext.succeeding(buffer -> testContext.verify(() -> {
-          assertThat(buffer.toString()).isEqualTo("Ok");
-          responsesReceived.flag();
-        })));
-    }
+        HttpClient client = vertx.createHttpClient();
+        for (int i = 0; i < 10; i++) {
+          client.request(HttpMethod.GET, 8888, "localhost", "/")
+            .compose(req -> req.send().compose(HttpClientResponse::body))
+            .onComplete(testContext.succeeding(buffer -> testContext.verify(() -> {
+              assertThat(buffer.toString()).isEqualTo("Ok");
+              responsesReceived.flag();
+            })));
+        }
+      }));
   }
 
   class CTest {

